blog/_posts/2022-02-24-rancher-desktop-container-runtimes.md

```diff
@@ -0,0 +1,118 @@
+image: "/assets/images/rancher-desktop-container-runtimes/title.jpg"
+image_caption: 'Photo by <a rel="noopener noreferrer" href="https://unsplash.com/photos/u0vgcIOQG08">Jens Lelie</a> on Unsplash'
+description: "Configure Tilt to build images with kim for use with Rancher Desktop"
+tags:
+- docker
+- moby
+- containerd
+- rancher
+- kubernetes
+---
+
+[Rancher Desktop][rancher-desktop], a lightweight and local Kubernetes cluster solution, has been evolving _fast_ over the past few months!
+
+Early releases used [containerd][] behind-the-scenes with an experimental CLI tool, [`kim`][kim], to build images.
+I wrote up a blog post about how easy it was to integrate with Tilt: [Writing Yet Another Custom Image Builder][blog-kim-extension].
+
+Since then, however, Rancher Desktop has started also supporting Docker (aka `dockerd (moby)`) as an alternative to containerd.
+Furthermore, [`kim`][kim] is no longer under active development.
+
+While this might seem like a lot of churn, none of this is necessarily a bad thing!
+In fact, when I was first writing about all this, our CTO, Nick, even predicted some of these changes.
+
+If you're currently a Docker Desktop user who's interested in checking out Rancher Desktop, you're also in luck:
+I've updated our [Switch from Docker Desktop to Rancher Desktop in 5 Minutes][blog-rancher-desktop-switch] blog post! 🎉
+
+While that post is a great place to start, let's take a peek at the container runtime options in a bit more detail.
+Why might you choose one over the other?
+How do they work with Tilt?
+
+### containerd
+The default container runtime in Rancher Desktop is [containerd][].
+
+These days, it's also the de facto standard container runtime used in production Kubernetes installs, so it's not surprising it's the default here!
+In fact, it's also used by [kind][] and in some [minikube][minikube-runtime] configurations among others.
+
+A popular option for building images with containerd is [nerdctl][] (a non-core subproject of containerd).
+As a bonus, `nerdctl` is drop-in compatible for the `docker` command.
+Luckily for us, Rancher Desktop even bundles a version of `nerdctl` already configured to build to its containerd instance.
+
+Be sure it's enabled by opening the Rancher Desktop preferences, navigating to **Supporting Utilities**, and checking the box for `nerdctl`.
+
+![Rancher Desktop preferences pane showing the nerdctl option](/assets/images/rancher-desktop-container-runtimes/nerdctl.png)
+
+Once that's done, we can use it with Tilt via the [`nerdctl` extension][ext-nerdctl].
+For example:
+```python
+# docker_build(
+#     ref='registry.example.com/my-image',
+#     context='.',
+# )
+# ⬇️ ⬇️ ⬇️ ⬇️
+load('ext://nerdctl', 'nerdctl_build')
+nerdctl_build(
+    ref='registry.example.com/my-image',
+    context='.',
+)
+```
+
+So, why are some reasons we would opt to use [containerd][] with Rancher Desktop?
+* You're using containerd in prod and want your dev environment to mirror that
```

Comment:
`containerd` is a smaller, faster, more secure container runtime

The trade-off is that `dockerd` has a lot of historical baggage, and wants to maintain backwards compatibility where it can. Some of that backwards compatibility is good and useful and you may have tools that depend on it!